cvrf2csaf/cvrf2csaf.py

```diff
@@ -56,7 +58,34 @@ def __init__(self, config):
             'Vulnerability': self.vulnerability,
         }
 
+    def _update_CVSSv3_version_from_schema(self, root_element):
+        """ Tries to update CVSS 3.x version from schema."""
+        cvss_3_regex = '.*cvss-v(3\.[01]).*'
+
+        potential_cvss3 = None
+
+        # iterate over  namespaces
+        for ns in root_element.nsmap.values():
+            match = re.match(cvss_3_regex, ns)
+            if match:
+                cvss_version_matched = match.groups()[0]
+                # no potential cvss version found yet -> store it
+                if not potential_cvss3:
+                    potential_cvss3 = cvss_version_matched
+                # already have some version, but it's the same as currently matched -> ok, continue
+                elif potential_cvss3 == cvss_version_matched:
+                    continue
+                # else we have two different potential cvss versions -> skip this step comletely
```

Comment:
I have to double check that. My first impression was that this could be valid:
```
<CVSSScoreSets>
      <ScoreSetV3 xmlns="https://www.first.org/cvss/cvss-v3.0.xsd>
        <BaseScoreV3>7.5</BaseScoreV3>
         </ScoreSetV3>
      <ScoreSetV3 xmlns="https://www.first.org/cvss/cvss-v3.1.xsd>
        <BaseScoreV3>7.5</BaseScoreV3>
         </ScoreSetV3>
      </CVSSScoreSets>
```

In that case, the element would clearly indicate its version...

Reply:
In that case the document doesn't pass input xsd validation:
`
Element '{https://www.first.org/cvss/cvss-v3.1.xsd}ScoreSetV3': This element is not expected. Expected is one of ( {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV2, {http://docs.oasis-open.org/csaf/ns/csaf-cvrf/v1.2/vuln}ScoreSetV3 )
`

Reply:
Well, that is not what was intended...

Here is my suggestion:
If such error occurs, we output a warning ("unexpected ScoreSetV3 element found. This might be fixed during the conversion. Proceeding for now.") and ignore it. Then, we proceed as the standard suggests (I'm splitting up the steps to be more precise):
1. Check the Vector first.
2. Determine it from the elements namespace. (Output a warning, if the namespace existed as discussed "CVSS version guess from elements namespace".)
3. Determine it from the global namespaces, but skip step if more than one exists. (Output a warning, if the namespace existed as discussed "CVSS version guess from global namespace".)
4. Config value + warning.

Reply:
Do I understand correctly that we should ignore these type of errors during xsd validation? So far we just stop the conversion right after failed validation of input document.

Reply:
Depends on the definition of "these type of errors". The `ScoreSetV3` needs a little special handling...

So the idea is: Only if a namespace error at any `ScoreSetV3` occurs, we output a warning (see above) and proceed. If there is any other error (including namespace errors in other elements), we stop the conversion right after the failed validation. If the namespace error from above occurs and any other error, we stop the conversion right after the failed validation. Does that make sense?

Reply:
@zpevma: @mfd2007 ran into a second case where such a "Workaround" is necessary... see #57

Reply:
Ok, I would create new issue + PR for this: 
 - allow some xsd validation errors (`ScoreSetV3` namespace error... )
 - try to determine CVSSv3.x version from `ScoreSetV3` schema
 - handle regex issues 

Do you agree to close this PR as is now? So far cvss version is determined like this:
 - first take version from config
 - if we can reliably determine version from global namespaces, forget the config one and use this one from NS
 - if we can determine version from vector, use this one

Reply:
@zpevma Agree to merge this in the current state and solve the rest via a separate PR.